### ISSUE TITLE
Not working (r/w error)

### ISSUE THREAD

**Issue by ajkblue** [da2x/fluxfonts#4]
I have had FluxFonts installed for a while now, but I cannot tell if it's working, but I don't believe that it is. I have checked [Browserleak's font tester webpage](http://www.browserleaks.com/fonts) hoping to be able to see a difference in the visible fonts after a while, but it's never changed. I believe that the program isn't working correctly, as searching for "fluxfontd" in Console shows multiple lines of this:

```
7/15/15 12:22:22.665 PM fluxfontd[2976]: Neither the `/run/lock/` nor `/var/lock/` directories are read-writeable!
```

I believe that that is the problem (I'm on OS X Yosemite 10.10.4) as to why it's not working. I got the error above after installing the package from [your webpage](https://www.aeyoun.com/projects/fluxfonts/index.html), which states itself as version 1.1. But since that doesn't seem to work, I then tried to build it from source and ran the fluxfontd again and I received this message (almost identical):

```
7/15/15 2:08:14.003 PM fluxfontd[11860]: Neither the '/run/' nor '/var/run/' directories are read-writable! Exiting.
```

I assume that the program isn't working properly, and if is isn't how do I get it to work?
Also, what is the proper way to check that it is working?

**Comment by Aeyoun** [comment]
It looks like the installer is broken in Yosemite. You can work around it by starting fluxfonts by running it with sudo in the Terminal. Will fix the installer probably sometime next week. Thanks for informing me!

On Mac you can open /Applications/Font Book.app and keep an eye on your list of fonts. You’ll see the list of fonts change over time (random interval but at least a couple of times per hour). Killing the fluxfonts process and restarting it should have an immediate effect, for testing purposes. Note that your browser, OS, or plug-ins may cache the font list for some time so browser based test may indeed show the same result two times in a row when testing, but over an hour the caches will expire and you’ll have a new unique font fingerprint. Fluxfont makes font fingerprint unreproducible over time (killing it as a long-term super-cookie) but within the same hour, your fingerprint may stay the same only to change later.

**Comment by ajkblue** [comment]
Thank you for the quick response, but as I've been watching my font list in Font Book and Safari for changes since the last few hours I haven't see any differences. Also, I don't see any active process from fluxfonts to kill, so I just ran the "fluxfontd" program again as sudo in Terminal.
```
7/15/15 4:27:15.624 PM sudo[20869]:  austink : TTY=ttys000 ; PWD=/Users/austink ; USER=root ; COMMAND=/usr/local/bin/fluxfontd
7/15/15 4:27:15.629 PM fluxfontd[20871]: Neither the `/run/lock/` nor `/var/lock/` directories are read-writeable!
```
Should I be looking at the *total number* of fonts in Font Book to change (shown at the top of the window in Font Book) or for the *actual list* of fonts to change?
![fontbook](https://cloud.githubusercontent.com/assets/9651939/8711547/29d452e2-2b10-11e5-8de6-44c9baa2b43c.png)
When querying the fonts installed using [Browserleaks.com test](http://www.browserleaks.com/fonts), here is what I always get, even after clearing browser caches and history (in Safari):
![browserleaks.com](https://cloud.githubusercontent.com/assets/9651939/8711736/cd3e12d2-2b11-11e5-80fa-1b085809781c.png)
Basically, even after running the daemon with sudo I don't think it's working. I also noticed that I don't have access to view "fluxfonts.plist" in /Library/LaunchDaemons/ either, or is that intended?

**Comment by Aeyoun** [comment]
Just tested and retested and triple tested everything and it works in every recent and upcoming OS X version as far as I can tell. I didn’t use the installer, however. If you download the source, and write `make` in Terminal and then `sudo ./fluxfontd` afterwards, is it working for you? (Requires Xcode for free from the App Store.)

**Comment by ajkblue** [comment]
I just downloaded it here from source, built it (with ``make``), and ran ``sudo ./fluxfontd`` and it still doesn't work for me! Here is the output of ``make`` in Terminal (I'm on El Capitan Public Beta 3):
![screen shot 2015-07-30 at 3 48 04 pm](https://cloud.githubusercontent.com/assets/9651939/8996074/bbbba778-36d2-11e5-81fd-87b2e73ad0b5.png)
I took a screenshot of the Browserleaks font test webpage before and after running it, first clearing the browser cache, and quitting Safari. To check for a changed result I opened a new (regular) window and a new private window in Safari. Every time I get the exact same result. I checked both Font Book and the Browserleaks test, but the fonts shown and the total number didn't change. I have the latest version of [Xcode-beta (version 4)](https://developer.apple.com/xcode/downloads/) when I built it. I get this from Console now when running it:
```
7/30/15 3:45:04.865 PM sudo[5560]:  ajkblue : TTY=ttys000 ; PWD=/Users/ajkblue/Downloads/fluxfonts-master ; USER=root ; COMMAND=./fluxfontd
7/30/15 3:45:04.871 PM fluxfontd[5607]: Could not lock '/var/run/fluxfonts.pid'. It could be locked by another process (1741). Exiting.
```
What else should I try to get it working?

**Comment by Aeyoun** [comment]
> 7/30/15 3:45:04.871 PM fluxfontd[5607]: Could not lock '/var/run/fluxfonts.pid'. It could be locked by another process (1741). Exiting.

This means it’s already working properly.

Run `cat /var/lib/fluxfonts/fonts` (not sure if it must be run with sudo), then open Font Book.app and you’ll see fonts from the list produced by the previous file. Keep an eye on that file over time to see it generate new fonts.

Can you open Font Book and look for any odd

**Comment by ajkblue** [comment]
Yes! Running ``cat /var/lib/fluxfonts/fonts`` with sudo printed the fonts that fluxfontd generated, and they were also present in Font Book. So it appears to be working! Except for I can't use ``make install`` as it throws me this error:
```
Secure-MBA:fluxfonts-master ajkblue$ make install-all
cc -liconv -Wall -std=c99 -g -arch x86_64 -arch i386 main.c lib/buffer.c lib/familyname.c lib/opentype.c lib/utils.c lib/tables/cff.c lib/tables/cmap.c lib/tables/head.c lib/tables/hhea.c lib/tables/hmtx.c lib/tables/maxp.c lib/tables/name.c lib/tables/os2.c lib/tables/post.c lib/tables/tables.c -o fluxfontd -lm
/usr/bin/install --directory /usr/local/bin
/usr/bin/install: illegal option -- -
usage: install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 file2
       install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 ... fileN directory
       install -d [-v] [-g group] [-m mode] [-o owner] directory ...
make: *** [install] Error 64
```
I then tried to install the launchd and that didn't appear to work either:
```
Secure-MBA:fluxfonts-master ajkblue$ make install-launchd
rm /private/var/tmp/fluxfonts.dist/dest_root/Library/LaunchDaemons/fluxfonts.plist
rm: /private/var/tmp/fluxfonts.dist/dest_root/Library/LaunchDaemons/fluxfonts.plist: No such file or directory
make: [configure-launchd] Error 1 (ignored)
/usr/bin/install --directory /private/var/tmp/fluxfonts.dist/dest_root/Library/LaunchDaemons
/usr/bin/install: illegal option -- -
usage: install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 file2
       install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 ... fileN directory
       install -d [-v] [-g group] [-m mode] [-o owner] directory ...
make: *** [configure-launchd] Error 64
```
That's the right way to install it on OS X, right?

**Comment by Aeyoun** [comment]
`sudo make install && sudo make install-launchd`

I’ll try to make this more user-firendly and better documented some day soon. Sorry for my poor communication on how to use this little utility of mine.

**Comment by ajkblue** [comment]
Still doesn't seem to install using those commands, either:
```
Secure-MBA:fluxfonts-master ajkblue$ sudo make install && sudo make install-launchd
Password:
cc -liconv -Wall -std=c99 -g -arch x86_64 -arch i386 main.c lib/buffer.c lib/familyname.c lib/opentype.c lib/utils.c lib/tables/cff.c lib/tables/cmap.c lib/tables/head.c lib/tables/hhea.c lib/tables/hmtx.c lib/tables/maxp.c lib/tables/name.c lib/tables/os2.c lib/tables/post.c lib/tables/tables.c -o fluxfontd -lm
/usr/bin/install --directory /usr/local/bin
/usr/bin/install: illegal option -- -
usage: install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 file2
       install [-bCcpSsv] [-B suffix] [-f flags] [-g group] [-m mode]
               [-o owner] file1 ... fileN directory
       install -d [-v] [-g group] [-m mode] [-o owner] directory ...
make: *** [install] Error 64
```

**Comment by da2x** [comment]
Resolved by commit cfd366a.

**Comment by ajkblue** [comment]
@da2x Better late than never! I'm glad to see this working again.
